### PR TITLE
Merge metadata into object's data

### PR DESCRIPTION
**Pull request by harenson**
After KZOO-248 changes, some Monster-UI applications were broken because the information they use to know whether or not a user/account has X capability used to be included within the actual object's data (response.data), but now some of that  information is being included outside of the object's data within a new field called `metadata` (response.metadata).

This change aims to fix this issue by merging `response.metadata` object into `response.data` object, every time a _non-listing_ request's response is received.

### CODE REVIEW

File: src/js/lib/jquery.kazoosdk.js [hunk 1]
```diff
@@ -780,6 +787,14 @@
 			var postData = data.data,
 				envelopeKeys = {};
 
+			// Remove object's metadata (if any) merged in the success step.
+			if (postData.hasOwnProperty('metadata')) {
+				_.each(postData.metadata, function(index, name) {
+					name !== 'id' && delete postData[name];
+				});
+				delete postData.metadata;
```
Comment (azefiel) [inline src/js/lib/jquery.kazoosdk.js:795]: Similarly here, we can use jQuery's `each()` instead:
```suggestion
				$.each(postData.metadata, function(key) {
					if (key === 'id') {
						return;
					}
					delete postData[key];
				});
```

**Review by azefiel — COMMENTED** [review]
Would there be a way to have a list of all keys that could make it into `metadata`?

That would make it easier for search and replace later.

**Comment by harenson** [comment]
@azefiel Per @jamesaimonetti "What metadata is returned will depend on the doc, like account docs leak a bunch of fields, while others may not leak any extra."

File: src/js/lib/jquery.kazoosdk.js [hunk 2]
```diff
@@ -764,6 +764,13 @@
 		};
 
 		settings.success = function requestSuccess(responseData, status) {
+			if ('object' == typeof responseData.data && responseData.hasOwnProperty('metadata')) {
+				// Merge object's metadata with the actual object's data.
+				responseData.data = _.merge(responseData.data, responseData.metadata);
```
[the remark this answers is not in the document]
Comment (harenson) [reply src/js/lib/jquery.kazoosdk.js]: @azefiel response for your question about possibly data and metadata duplicated keys: Only _id_ field is duplicated in both objects, I do not see any other field being duplicated at the moment.

[the remark this answers is not in the document]
Comment (harenson) [reply src/js/lib/jquery.kazoosdk.js]: Regarding _deep merge_ when using `$.extend`, there is not need for it, as none of the leaked fields are objects. Also, field duplication is not expected, there should not be a try to merge a non-object value with an object value.

Why not mutate if that is actually what this line is doing, it is setting `responseData.data` equals to the output of the merge action, so it was simplified.

[the remark this answers is not in the document]
Comment (harenson) [reply src/js/lib/jquery.kazoosdk.js]: For you information, you mention this library should not rely on 3rd party libraries, but I can see `_.` (Lodash) is being used in other places within this same file.

**Comment by pcandia** [comment]
@harenson I don't think these changes are now relevant. we have made changes in some of the apps where the changes to the API were made (5.2) and not all of them have the same structure. but after the changes per app that we made they are working as expected.  Sorry, I didn't see this PR before.

**Comment by harenson** [comment]
No worries. Thank you for taking care of it.

**Pull request closed without merging**